inc/class-core-sitemaps-provider.php

```diff
@@ -42,6 +42,23 @@ class Core_Sitemaps_Provider {
 	 */
 	public $slug = '';
 
+	/**
+	 * Set up relevant rewrite rules, actions, and filters.
+	 */
+	public function setup() {
```

Comment:
I've moved the setup for all actions performed by providers from the main sitemap class and into the provider itself to keep responsibilities clear.